Library/Homebrew/os/linux/elf.rb

```diff
@@ -127,7 +136,22 @@ def initialize(path)
     private
 
     def needed_libraries(path)
-      if DevelopmentTools.locate "readelf"
+      if HOMEBREW_PATCHELF_RB
+        patcher = path.patcher
+        soname = begin
+            patcher.soname
+          rescue PatchELF::PatchError # => e
+            # opoo e
```

Comment:
@sjackman the tests expect no output to stderr. shall we proceed as it is without logging ?

Reply:
Fix line 142 so that `patcher.soname` is only called when `path.dylib?`